eng/scripts/Generate-Patch.ps1

```diff
@@ -120,42 +106,45 @@ function UpdateChangeLog($ArtifactName, $ServiceDirectoryName, $Version) {
   Set-ChangeLogContent -ChangeLogLocation $ChangelogPath -ChangeLogEntries $ChangeLogEntries
 }
 
-if('' -eq $PatchVersion) {
-  $PatchVersion = GetPatchVersion -ReleaseVersion $ReleaseVersion
-  if('' -eq $PatchVersion) {
-    LogError "Could not fetch the patch version. Exitting ..."
-    exit
+function ResetSourcesToReleaseTag($ArtifactName, $ServiceDirectoryName, $ReleaseTag, $RepoRoot, $RemoteName) {
+  $SdkDirPath = Join-Path $RepoRoot "sdk"
+  $ServiceDirPath = Join-Path $SdkDirPath $ServiceDirectoryName
+  $ArtifactDirPath = Join-Path $ServiceDirPath $ArtifactName
+  $ArtifactPomFile = Join-Path $ArtifactDirPath "pom.xml"
+  
+  TestPathThrow -Path $ArtifactDirPath
+  TestPathThrow -Path $ArtifactPomFile
+  
+  $ResetSources = $true
+  
+  try {
+    [xml]$PomFileContent = Get-Content -Path $ArtifactPomFile
+    $CurrentVersion = $PomFileContent.project.version
+    if($CurrentVersion -eq $ReleaseVersion) {
+      $ResetSources = $false
+    }
   }
-}
-Write-Information "PatchVersion is: $PatchVersion"
-
-$RemoteName = GetRemoteName -MainRemoteUrl $MainRemoteUrl
-if($null -eq $RemoteName) {
-    LogError "Could not fetch the remote name for the URL $MainRemoteUrl Exitting ..."
-    exit
-}
-Write-Information "RemoteName is: $RemoteName"
-
-if('' -eq $BranchName) {
-  $ArtifactNameToLower = $ArtifactName.ToLower()
-  $BranchName = "release/$ArtifactNameToLower/$PatchVersion"
-}
-Write-Information "BranchName is: $BranchName"
+  catch {
+    # Could not parse the results.
+  }
+  
+  if($ResetSources -eq $false) {
+    return
+  }
+  
+  $TestResourcesFilePath = Join-Path $ServiceDirPath "test-resources.json"
+  $EngDir = Join-Path $RepoRoot "eng"  
+  $CodeQualityReports = Join-Path $EngDir "code-quality-reports" "src" "main" "resources"
+  $CheckStyleSuppressionFilePath = Join-Path $CodeQualityReports "checkstyle" "checkstyle-suppressions.xml"
```

Comment:
For Checkstyle you may need to reset the configuration in addition to the suppressions as the configurations could introduce new rules